### ISSUE TITLE
[cryptolib] Correct flag0 values for CSRNG and EDN configurations

### ISSUE THREAD

**Issue by vogelpi** [lowRISC/opentitan#22524]
### Description

While debugging ENTROPY_SRC/CSRNG test failures, it has been discovered that CSRNG triggeres a recoverable alert and sets the ACMD_FLAG0_FIELD_ALERT alert status bit during ROM_ext.

The reason for this is most likely that cryptolib doesn't configure the EDNs correctly. More precisely, inside `entropy_complex_init()` which is called in ROM_ext, the entropy complex is configured with the settings defined in `kEntropyComplexConfigIdContinuous` (see `entropy.c`).

I am not exactly sure where the misconfiguration happens. I've found two things so far:

1. The configuration struct has configurations for both EDN0 and EDN1. For EDN1, there the `disable_trng_input` struct member for the Generate command is not defined (it's defined for the EDN0 Generate command though.):
```c
                .edn1 =
                    {
                        .base_address = kBaseEdn1,
                        .reseed_interval = 4,
                        .instantiate =
                            {
                                .id = kEntropyDrbgOpInstantiate,
                                .disable_trng_input = kHardenedBoolFalse,
                                .seed_material = NULL,
                                .generate_len = 0,
                            },
                        .generate =
                            {
                                .id = kEntropyDrbgOpGenerate,
                                .seed_material = NULL,
                                .generate_len = 1,
                            },
                        .reseed =
                            {
                                .id = kEntropyDrbgOpReseed,
                                .disable_trng_input = kHardenedBoolFalse,
                                .seed_material = NULL,
                                .generate_len = 0,
                            },
                    },
```
2. `ENTROPY_CMD` macro leaves the flag0 field for the register empty unless `kHardenedBoolTrue` true is provided via the above struct.
```c
  if (launder32(cmd.disable_trng_input) == kHardenedBoolTrue) {
    reg = bitfield_field32_write(reg, kAppCmdFieldFlag0, kMultiBitBool4True);
  }
```
So 0 is written which is not a valid MuBi4Value expected by CSRNG.

**Comment by vogelpi** [comment]
FYI @jadephilipoom

**Comment by vogelpi** [comment]
The problem here is twofold:
1. `edn_configure` just sets up EDN. From that point it takes a while (depending on hardware entropy consumers) until the "faulty" flag0 gets transmitted into CSRNG where it triggers the error. The instantiate command is sent almost immediately, but it may take a very long time for the `Reseed` to be sent.
2. The old EDN / CSRNG versions in ES, the command acknowledgement and status reporting was only partially working. @h-filali has been working hard to fix this.

**Comment by ballifatih** [comment]
I was reading entropy configuration code this week, and it was bothering me that there is not an explicit default value for flag0 (in entropy.c). After going back and forth between C and RTL, my expectation was that an alert was supposed to be fired for SW-controlled CSRNG instance. I was not looking at provisioning code, but not ROM_ext, so I thought maybe we ignore alerts during provisioning.

I do not see an issue with the config structure, since generate operation is not supposed to consume seed anyway and flag0 controls the construction of the seed in RTL.

I think the second reason you suggested is the root cause @vogelpi. Is it possible to test the fix from #22529?

**Comment by vogelpi** [comment]
I think you're right @ballifatih , the config structure should be fine. The flag0 value is only latched into a register and then checked for Instantiate and Reseed commands.
